telegram/ext/_chatjoinrequesthandler.py

```diff
@@ -65,4 +101,12 @@ def check_update(self, update: object) -> bool:
             :obj:`bool`
 
         """
-        return isinstance(update, Update) and bool(update.chat_join_request)
+        if isinstance(update, Update) and update.chat_join_request:
+            chat = update.chat_join_request.chat
+            if self._chat_ids and chat.id not in self._chat_ids:
+                return False
```

Comment:
consider a chat with username `@A` and id 1. if `'A' in `self._usernames`, but `1 not in `self._chat_ids`, the request will be rejected. Is this intended? CC @Poolitzer 

Whatever the decision is, I would ask to explicitly clarify that in the docs and also add a test case for that.

Reply:
I would argue to implement it like an or filter. If either the username or the chat id correspond to one in appropriate set, set it to `True`; otherwise `False`. So change from current behavior.

But yes, documenting this clearly would be great.